File: backend/app/models/book.rb
class Book < ApplicationRecord
  enum :status, { unread: "unread", reading: "reading", completed: "completed" }

  validates :title, presence: true
  validates :author, presence: true
  validates :rating, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 5 }, allow_nil: true
end

File: frontend/src/types/book.ts
export type BookStatus = 'unread' | 'reading' | 'completed'

export interface Book {
  id: number
  title: string
  author: string
  status: BookStatus
  isbn?: string | null
  cover_image_url?: string | null
  rating?: number | null
  memo?: string | null
  started_at?: string | null
  completed_at?: string | null
  created_at?: string
}
